src/views/FacilityDetails.vue

```diff
@@ -959,7 +964,45 @@ export default defineComponent({
       } catch(err) {
         logger.error(err)
       }
-    }
+    },
+    async renameFacility() {
+      const alert = await alertController.create({
+        header: translate("Rename facility"),
+        inputs: [{
+          name: "facilityName",
+          value: this.current.facilityName
+        }],
+        buttons: [{
+          text: translate('Cancel'),
+          role: "cancel"
+        },
+        {
+          text: translate('Apply'),
+          handler: async (data: any) => {
+            if(data.facilityName) {
+              try {
+                const resp = await FacilityService.updateFacility({
+                  facilityId: this.facilityId,
+                  facilityName: data.facilityName
+                })
+
+                if (!hasError(resp)) {
+                  showToast(translate("Facility renamed successfully."))
+                  await this.store.dispatch('facility/fetchCurrentFacility', { facilityId: this.facilityId })
```

Comment:
Updating facility name does not update the UI when the uses comes from the list page and then directly change the facility name as in that case no api call will be made and we will not fetch the same information for the facility again if the same facilityId is found in the list.
Instead update the name in the current object and then update the state locally on api success.